course7/homeworks/kushagra/hw2/ls_hw2.cpp

```diff
@@ -0,0 +1,169 @@
+        spaces += "    ";
+        for (fs::directory_entry& x : fs::directory_iterator(display_path)) {
+            print_dir_contents(x.path(), spaces);
+        } 
+    } 
+}
+
+
+/*
+Calculate the size of the files in the directory. If recursive is set to true, the total size of all files in the
+directory and its subdirectories. If set to false, only the size of the files in the directory is calculated and
+returned.
+*/
+int get_dir_size(const fs::path& dir_path, bool recursive=false) {
+    int dir_size = 0;
+	if(recursive) {
+		for (fs::directory_entry& x : fs::recursive_directory_iterator(dir_path)) {
+			if (fs::is_regular_file(x)) {
+				dir_size += fs::file_size(x);
+			}
+		}
+	} else {
+		for (fs::directory_entry& x : fs::directory_iterator(dir_path)) {
+			if (fs::is_regular_file(x)) {
+				dir_size += fs::file_size(x);
+			}
+		}
+	}
+	return dir_size;
+}
+
+
+/*Display the directory contents, along with the sizes. The size of the directory is calculated by the sum of all
+the sizes of immediate descendant files in the directory
+*/
+void print_detailed_dir_contents(const fs::path& display_path) {
+	int file_size = 0;
+	int dir_size = get_dir_size(display_path);
+	std::cout << std::to_string(dir_size) << "    " << display_path.filename().generic_string()
+			  << fs::path::preferred_separator << std::endl;
+	for (fs::directory_entry& x : fs::directory_iterator(display_path)) {
+		if (fs::is_regular_file(x.path())) {
+			file_size = fs::file_size(x.path());
+			std::cout << "    " << std::to_string(file_size) << "    " << x.path().filename().string() << std::endl;
+		}
+		else {
+			dir_size = get_dir_size(x.path());
+			std::cout << "    " << std::to_string(dir_size) << "    " << x.path().filename().generic_string()
+					  << fs::path::preferred_separator << std::endl;
+		}
+    }
+}
+
+
+/*Display the directory contents, along with the sizes. The size of the directory is calculated by the sum of all
+the sizes of descendant files in the directory, as well as its subdirectories.
+*/
+void print_detailed_dir_contents(const fs::path& display_path, std::string& spaces) {
+	int dir_size = 0;
+	int file_size = 0;
+	if (fs::is_regular_file(display_path)) {
+		file_size = fs::file_size(display_path);
+		std::cout << spaces << std::to_string(file_size) << "    " << display_path.filename().string() << std::endl;
+	}
+    else {
+		dir_size = get_dir_size(display_path, true);
+        std::cout << spaces << std::to_string(dir_size) << "    " << display_path.filename().generic_string()
+				  << fs::path::preferred_separator << std::endl;
+        spaces += "    ";
+        for (fs::directory_entry& x : fs::directory_iterator(display_path)) {
+            print_detailed_dir_contents(x.path(), spaces);
+        }
+    }
+}
+
+
+int main(int argc, char** argv)
+{
+	std::string path_name = ".";
+	bool details = false, recursive = false;
+	
+    po::options_description desc("Display contents of a directory.");
```

Comment:
Inconsistent indentation

Reply:
Indeed. I need to fix this indentation, I think it's coming from me doing half of the homework at home, and other at the office.